.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1007,23 +1007,12 @@ private DateTimeResolutionResult ParseOneWordPeriod(string text, DateObject refe
                         if (!string.IsNullOrEmpty(match.Groups["special"].Value))
                         {
                             var specialYearTimex = this.config.SpecialYearPrefixesMap[match.Groups["special"].Value.ToLowerInvariant()];
-
-                            if (!string.IsNullOrWhiteSpace(specialYearTimex))
-                            {
-                                swift = this.config.GetSwiftYear(trimmedText);
-                                if (swift < -1)
-                                {
-                                    ret.Timex = Constants.TimexFuzzyYear + specialYearTimex;
-                                }
-                                else
-                                {
-                                    var yearStr = year.ToString("D4");
-                                    ret.Timex = yearStr + specialYearTimex;
-                                }
-
-                                ret.Success = true;
-                                return ret;
-                            }
+                            swift = this.config.GetSwiftYear(trimmedText);
+                            ret.Timex = swift < -1 ?
+                                TimexUtility.GenerateYearTimex() + specialYearTimex :
```

Comment:
specialYearTimex is part of the timex generation.
And if swift changes the timex, than it should also be passed to the timex builder.